hazelcast/src/main/java/com/hazelcast/internal/networking/AbstractChannel.java

```diff
@@ -79,14 +87,18 @@ public SocketChannel socketChannel() {
 
     @Override
     public SocketAddress remoteSocketAddress() {
-        Socket socket = socket();
-        return socket == null ? null : socket.getRemoteSocketAddress();
+        if (remoteAddress == null) {
```

Comment:
Nulll check isn't needed because a socketchannel.socket will never return null.

```
 public Socket socket() {
        Object var1 = this.stateLock;
        synchronized(this.stateLock) {
            if (this.socket == null) {
                this.socket = SocketAdaptor.create(this);
            }

            return this.socket;
        }
    }
```